### PR TITLE
Ensure watcher can wait on run folder to appear

### LINKED ISSUE #377: Ensure watcher can run even if run folder not written to Data
**Is your feature request related to a problem? Please describe.**

Currently, the watcher can only start if the folder for a particular run has been written to the MIBIData folder, which only happens after the first FOV is generated. The watcher should run even if this folder has not been created.

**Describe the solution you'd like**

Provide a separate listener to watch for the creation of the run folder, then spin up the watcher instance once this gets created.

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Closes #377. The watcher is not currently designed to run until the run folder appears. This is problematic because this doesn't happen until about 15-30 minutes into the start of a run. The user needs to be able to spin up the watcher immediately after the run starts, otherwise it causes a significant inconvenience.

**How did you implement your changes**

`fov_watcher.start_watcher` will now take a new argument, `run_folder_timeout`, that controls how long the watcher waits for the run folder to appear. This allows implementation of timing logic similar to the timeout logic for waiting on `.bin` and `.json` files. The watcher will wait until either the run folder appears (in which case it listens for files inside run folder) or `run_folder_timeout` is reached (in which case it errors out with an informative message to the user).

Occasionally, what happens is the CACs will change the casing and/or add extraneous characters (notably `"T"` as a timestamp marker) to run names. If the run folder specified by the user hasn't already written to the CAC, warn the user to double check the run folder doesn't already exist in the `D:\\Data` folder under a slightly different name.

**Remaining issues**

Should we instead include the warning in a cell prior?

### CODE REVIEW

File: src/toffy/fov_watcher.py [hunk 1]
```diff
@@ -540,12 +541,38 @@ def start_watcher(
...
+    if not os.path.exists(run_folder):
+        warnings.warn(
+            f"Waiting for {run_folder}. Please first double check that your run data "
+            "doesn't already exist under a slightly different name in D:\\Data. "
+            "Sometimes, the CACs change capitalization or add extra characters to the run folder. "
+            "If this happens, stop the watcher and update the run_name variable in the notebook "
+            "before trying again."
+        )
+
+    # allow the watcher to
```
Comment (srivarra) [inline src/toffy/fov_watcher.py:563]: The suspense! what is it allowed to do? 🤔

File: tests/fov_watcher_test.py [hunk 2]
```diff
@@ -127,6 +127,103 @@ def test_run_structure(run_json, expected_files, recwarn):
...
+        # ensure warn_overwrite set to False if intermediate callbacks set, otherwise True
+        kwargs["warn_overwrite"] = True if int_cbs else False
+
+        run_folder = os.path.join(tmpdir, "test_run")
+        log_out = os.path.join(tmpdir, "log_output")
+        fov_callback, run_callback, intermediate_callback = build_callbacks(
+            run_cbs, int_cbs, fov_cbs, **kwargs
+        )
+
+        with Pool(processes=4) as pool:
```
Comment (srivarra) [inline tests/fov_watcher_test.py:189]: Any particular reason to use 4 processes?

**Review by srivarra — CHANGES_REQUESTED** [review]
Looks good, have you tried this on a run yet?

File: src/toffy/fov_watcher.py (hunk 1 above)
Comment (alex-l-kong) [reply src/toffy/fov_watcher.py:563]: I'm giving the developer an opportunity to use their imagination 🤣

File: tests/fov_watcher_test.py (hunk 2 above)
Comment (alex-l-kong) [reply tests/fov_watcher_test.py:189]: Consistency with the main watcher test below. We need multiple to ensure these functions can run in parallel.

**Comment by alex-l-kong** [comment]
> Looks good, have you tried this on a run yet?

Yup, I tried it on Mako's

File: tests/fov_watcher_test.py [hunk 3]
```diff
@@ -127,6 +127,103 @@ def test_run_structure(run_json, expected_files, recwarn):
...
+                            intermediate_callback,
+                            10,
+                            1,
+                            SLOW_COPY_INTERVAL_S,
+                        ),
+                    )
+
+                    res_scan.get()
+            else:
+                res_scan = pool.apply_async(
```
Comment (camisowers) [inline tests/fov_watcher_test.py:210]: Just for test timing, I'm wondering if we need to see this watcher test to completion? Or is it sufficient to just exit out of start_watcher once the run folder has been created successfully and the script has successfully began to create outputs from the first few callbacks? 
If the watcher tests don't take that long then this is probably fine as is.

**Review by camisowers — COMMENTED** [review]
Looks good, just a small test efficiency thought. Also the builds/tests are lagging below.

File: tests/fov_watcher_test.py (hunk 3 above)
Comment (srivarra) [reply tests/fov_watcher_test.py:210]: Looks like the workflow isn't starting the test process, but I am able to manually run it with.

**Review by srivarra — APPROVED** [review]
Looks Good!

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**